src/system.rs

```diff
@@ -220,10 +248,27 @@ impl PendingTypes {
                 TypeDefinition::Named(named_type_def) => {
                     match isl_imported_type_name {
                         None => {
+                            // check if the type definitions that are not yet resolved actually exists within the schema
+                            // we can use the isl_type_names to make sure if they exists, otherwise return error.
+                            if named_type_def.is_deferred_type_def()
+                                && !isl_type_names
+                                    .contains(named_type_def.name().as_ref().unwrap().as_str())
+                            {
+                                return unresolvable_schema_error(
+                                    "Unable to load schema due to unresolvable type",
```

Comment:
suggestion–in the error message, include the name of the type (and _possibly_ the ID of the schema that we're attempting to import from) that cannot be resolved